Sources/TuistCore/Graph/GraphDependencyReference.swift

```diff
@@ -101,7 +102,7 @@ public enum GraphDependencyReference: Equatable, Comparable, Hashable {
             return lhsPath < rhsPath
         case let (.library(lhsPath, _, _, _), .library(rhsPath, _, _, _)):
             return lhsPath < rhsPath
-        case let (.product(lhsTarget, lhsProductName), .product(rhsTarget, rhsProductName)):
+        case let (.product(lhsTarget, lhsProductName, _), .product(rhsTarget, rhsProductName, _)):
```

Comment:
I'd take into account the new attribute for sorting two targets that have the same name but different product platform filter.